src/pipeline.rs

```diff
@@ -1,18 +1,38 @@
 use std::fs::File;
 use std::io::{BufRead, BufReader};
 
-use crate::configuration::Configuration;
+use crate::configuration::{Column, Configuration, FileType};
 use crate::embedding::{calculate_embeddings, calculate_embeddings_mmap};
 use crate::entity::{EntityProcessor, SMALL_VECTOR_SIZE};
 use crate::persistence::embedding::TextFileVectorPersistor;
 use crate::persistence::entity::InMemoryEntityMappingPersistor;
 use crate::persistence::sparse_matrix::InMemorySparseMatrixPersistor;
 use crate::sparse_matrix::{create_sparse_matrices, SparseMatrix};
 use bus::Bus;
-use smallvec::SmallVec;
+use json;
+use smallvec::{smallvec, SmallVec};
 use std::sync::Arc;
 use std::thread;
 
+
+/// Parse a line of JSON and read its columns into a vector for processing.
+fn read_json_columns(line: &str, columns: &Vec<Column>) -> Vec<SmallVec<[String; SMALL_VECTOR_SIZE]>> {
+    let mut parsed = json::parse(&line).unwrap();
```

Comment:
`unwrap` and `expect("values for non-complex columns must be strings")` is better to log to don't waste those lines which are ok.

Reply:
@piobab Could you clarify what you mean here?

Reply:
I mean that parsing invalid line can stop whole processing. Instead of unwrap (will panic when json can't be parsed) you can log this line in case of error.

Reply:
This is already the case for the existing TSV implementation so I kept the same behavior.

Reply:
I guess in the future we could implement a flag that would either stop on error or skip errors. Also probably error message of unwrap wouldn't be very useful. However, I think it would be good to prepare such changes in separate PR for both CSV and JSON formats.

Reply:
@michaelmior @kodieg sure - we will fix it in separate PR. It is my oversight that there is no invalid line support.